File: client/lib/features/transactions/domain/entities/transaction.dart
class Transaction {
  final String userId;
  final int jarId;
  final String type;
  final double amount;
  final String description;
  final DateTime transactionDate;

  Transaction({
    required this.userId,
    required this.jarId,
    required this.type,
    required this.amount,
    required this.description,
    required this.transactionDate,
  });
}

File: server/src/domain/model/transaction.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type TransactionType string

const (
	Income  TransactionType = "income"
	Expense TransactionType = "expense"
)

type Transaction struct {
	ID              int             `json:"id"`
	UserID          uuid.UUID       `json:"user_id"`
	JarID           int             `json:"jar_id"`
	Type            TransactionType `json:"type"`
	Amount          float64         `json:"amount"`
	Description     string          `json:"description"`
	TransactionDate time.Time       `json:"transaction_date"`
	CreatedAt       time.Time       `json:"created_at"`
}
